src/3d/qgs3dsceneexporter.cpp

```diff
@@ -314,6 +315,14 @@ QgsTerrainTileEntity *Qgs3DSceneExporter::getFlatTerrainEntity( QgsTerrainEntity
   // the entity we created will be deallocated once the scene exporter is deallocated
   Qt3DCore::QEntity *entity = flatTerrainLoader->createEntity( this );
   QgsTerrainTileEntity *tileEntity = qobject_cast<QgsTerrainTileEntity *>( entity );
+
+  Qgs3DMapSettings *settings = terrain->mapSettings();
```

Comment:
We already have this in parseTerrain -- I think it should be added as an argument to getDemTerrainEntity/getFlatTerrainEntity/getMeshTerrainEntity instead.